tests/unit/basic_crawler/test_basic_crawler.py

```diff
@@ -671,6 +679,20 @@ async def handler(context: BasicCrawlingContext) -> None:
     assert (await store.get_value('foo')) == 'bar'
 
 
+async def test_context_use_state(mock_event_manager: EventManager) -> None:
+    crawler = BasicCrawler()
+
+    @crawler.router.default_handler
+    async def handler(context: BasicCrawlingContext) -> None:
+        await context.use_state('state', {'hello': 'world'})
+        await asyncio.sleep(0.1)
+
+    await crawler.run(['https://hello.world'])
+
+    store = await crawler.get_key_value_store()
+    assert (await store.get_value('state')) == {'hello': 'world'}
```

Comment:
I'd suggest adding a more complex test as well, perhaps one that accesses the shared state in different request handler "runs"?

Reply:
That's a great idea. So I found out that I didn't take into account that in different handlers we work with different `KeyValueStore` objects.

I moved the `_cache` to the class variable level. But maybe the best solution would be to make some kind of `CacheService`.